src/backend/distributed/Makefile

```diff
@@ -19,6 +19,12 @@ SUBDIRS = . commands connection ddl deparser executor master metadata planner pr
 OBJS += \
 	$(patsubst $(citus_abs_srcdir)/%.c,%.o,$(foreach dir,$(SUBDIRS), $(sort $(wildcard $(citus_abs_srcdir)/$(dir)/*.c))))
 
+# make sure citus_version.o is always recompiled to reflect the correct gitref for every build
+# idea came from https://stackoverflow.com/a/3304967
+utils/citus_version.o: FORCE
```

Comment:
Small downside is that `make install` now rebuilds citus_version.o and citus.so twice. I tend to do `make -j 8 && make install`, which rebuilds it 3 times.

What if we add a rule like:
`utils/citus_version.o: $(OBJS)`

That way, we rebuild it whenever a .o file changes.

I believe it needs to be below `include $(citus_top_builddir)/Makefile.global`.

(Bonus points for removing the circular dependency, though `make` does not seem to mind to much)